DistributionModels/weibull.py

```diff
@@ -84,27 +107,67 @@ def wscore(self, distances):
+        else:
+            return weibulls.cdf(distances)        
+
 
+        
+
+    def prob(self, distances):
+        """
+        This function can calculate raw probability scores from various weibulls for a given set of distances
+        :param distances: a 2-D tensor with the number of rows equal to number of samples and number of columns equal to number of weibulls
+        Or
+        a 1-D tensor with number of elements equal to number of test samples
+        :return:
+        """
+        self.deviceName = distances.device
+        scale_tensor = self.wbFits[:,1]
+        shape_tensor = self.wbFits[:, 0]
+        if self.sign == -1:
+            distances = -distances
+        if len(distances.shape)==1:
+            distances = distances.repeat(shape_tensor.shape[0],1)
+        smallScoreTensor=self.smallScoreTensor
+        if len(self.smallScoreTensor.shape)==2:
+            smallScoreTensor=self.smallScoreTensor[:,0]
+            ## tb hack 
+        distances = distances + self.translateAmountTensor - smallScoreTensor.to(self.deviceName)[None,:]
+#        distances = distances + 1 - smallScoreTensor.to(self.deviceName)[None,:]
+        weibulls = torch.distributions.weibull.Weibull(scale_tensor.to(self.deviceName),shape_tensor.to(self.deviceName))
+        distances = distances.clamp(min=0)
+        return torch.exp(weibulls.log_prob(distances))
+    
 
     def _weibullFitting(self, dataTensor, tailSize, isSorted=False, gpu=0):
         self.deviceName = dataTensor.device
-        if isSorted:
-            sortedTensor = dataTensor
-        else:
-            if self.sign == -1:
-                dataTensor = -dataTensor
-            sortedTensor = torch.topk(dataTensor, tailSize, dim=1, largest=True, sorted=True).values
+        if self.sign == -1:
+            dataTensor = -dataTensor
+
+#        if isSorted:
+#           could try to just pull top items but its complicated by
+#           sign and tesnor structor so for now just get topk 
+        sortedTensor = torch.topk(dataTensor, tailSize, dim=1, largest=True,
+                                      sorted=True).values
 
         smallScoreTensor = sortedTensor[:, tailSize - 1].unsqueeze(1)
-        processedTensor = sortedTensor + 1 - smallScoreTensor
+        #tb hask to remove shift by small score, its not needed with proper use of translateAmountTensor
+#        smallScoreTensor = 0*smallScoreTensor;
+        processedTensor = sortedTensor + self.translateAmountTensor - smallScoreTensor
+# previously translateAmountTensor was hard coded at 1. (translateamount from libMR)        
+#        processedTensor = sortedTensor + 1 - smallScoreTensor        
         # Returned in the format [Shape,Scale]
+        self.smallScoreTensor = smallScoreTensor
         if self.splits == 1:
             self.wbFits = self._fit(processedTensor)
-            self.smallScoreTensor = smallScoreTensor
+            return self.wbFits,smallScoreTensor
```

Comment:
Looks like a nice bug catch thanks @tboult, unsure if `self.smallScoreTensor` can be set outside in case `splits>1` @TouqeerAhmad could you please cross check